packages/shared/src/hooks/useReportComment.ts

```diff
@@ -0,0 +1,57 @@
+import { useContext } from 'react';
+import { useMutation } from 'react-query';
+import { graphqlUrl } from '../lib/config';
+import AuthContext from '../contexts/AuthContext';
+import { BooleanPromise } from '../components/filters/common';
+import { AuthTriggers } from '../lib/auth';
+import { useRequestProtocol } from './useRequestProtocol';
+import {
+  ReportCommentReason,
+  REPORT_COMMENT_MUTATION,
+} from '../graphql/comments';
+import { useToastNotification } from './useToastNotification';
+
+type UseReportCommentRet = {
+  reportComment: (variables: {
+    commentId: string;
+    reason: ReportCommentReason;
+    note?: string;
+  }) => BooleanPromise;
+  onReportCallback: () => Promise<void>;
+};
+
+interface ReportCommentProps {
+  commentId: string;
+  reason: ReportCommentReason;
+  note: string;
+}
+
+export default function useReportComment(): UseReportCommentRet {
+  const { user, showLogin } = useContext(AuthContext);
+  const { displayToast } = useToastNotification();
+  const { requestMethod } = useRequestProtocol();
+  const { mutateAsync: reportCommentAsync } = useMutation<
+    void,
+    unknown,
+    ReportCommentProps
+  >((variables) =>
+    requestMethod(graphqlUrl, REPORT_COMMENT_MUTATION, variables),
+  );
+
+  const onReportCallback = async (): Promise<void> => {
+    displayToast('🚨 Thanks for reporting!');
+  };
```

Comment:
Just wondering if we need this extracted here? I think it would be better to just call it where needed? Would love to hear what @rebelchris and @sshanzel think as well.

Reply:
Ideally actually we abstract the actual text in a common lib file so we only need to change it in one place.
Once that's done we can simply inline the toast yes.